File: Data Structures & Algorithms/sort-colors/submission-1.java
class Solution {
    public void sortColors(int[] nums) {
        int[] bucket = new int[nums.length + 1];
        for(int num : nums){
            bucket[num]++;
        }

        int index = 0;

        // [1, 2, 1]
        //  0  1  2
        for(int i = 0; i<bucket.length; i++){
            for(int j=0; j<bucket[i]; j++){
                nums[index] = i;
                index++;
            }
        }
    }
}

File: Data Structures & Algorithms/sort-an-array/submission-1.py
class Solution:
    def sortArray(self, nums: List[int]) -> List[int]:
        count = [0] * 500001

        for num in nums:
            count[num] += 1

        res = []

        for i in range(len(count)):
            while count[i] > 0:
                res.append(i)
                count[i] -= 1
        
        return res